packages/augur-sdk/src/state/db/BlockAndLogStreamerListener.ts

```diff
@@ -1,77 +1,122 @@
-  subscribeToOnLogsAdded: (onLogsAdded: (blockHash: string, logs: TLog[]) => void) => string;
-  subscribeToOnLogsRemoved: (onLogsRemoved: (blockHash: string, logs: TLog[]) => void) => string;
+  subscribeToOnBlockRemoved: (
+    onBlockRemoved: (block: TBlock) => void
+  ) => string;
+  subscribeToOnLogsAdded: (
+    onLogsAdded: (blockHash: string, logs: TLog[]) => void
+  ) => string;
+  subscribeToOnLogsRemoved: (
+    onLogsRemoved: (blockHash: string, logs: TLog[]) => void
+  ) => string;
 }
 
 export interface BlockAndLogStreamerListenerDependencies {
   address: string;
   blockAndLogStreamer: BlockAndLogStreamerInterface<Block, ExtendedLog>;
   getBlockByHash: (hashOrTag: string) => Promise<Block>;
   eventLogDBRouter: EventLogDBRouter;
-  getEventTopics: (eventName: string) => Array<string>;
+  getEventTopics: (eventName: string) => string[];
   // TODO Use an emitter?
   listenForNewBlocks: (callback: (block: Block) => Promise<void>) => void;
 }
 
 type GenericLogCallbackType<T, P> = (blockIdentifier: T, logs: P[]) => void;
 
-export type BlockstreamLogCallbackType = GenericLogCallbackType<string, Log>
-export type LogCallbackType = GenericLogCallbackType<number, ParsedLog>
+export type LogCallbackType = GenericLogCallbackType<number, ParsedLog>;
 
-export interface IBlockAndLogStreamerListener {
-  listenForEvent(eventName: string, onLogsAdded: LogCallbackType, onLogRemoved?: LogCallbackType): void;
+export interface BlockAndLogStreamerListenerInterface {
+  listenForEvent(
+    eventName: string,
+    onLogsAdded: LogCallbackType,
+    additionalTopics?: string[]
+  ): void;
   listenForBlockRemoved(callback: (blockNumber: number) => void): void;
   startBlockStreamListener(): void;
 }
 
-export class BlockAndLogStreamerListener implements IBlockAndLogStreamerListener {
+export class BlockAndLogStreamerListener
+  implements BlockAndLogStreamerListenerInterface {
   private address: string;
 
   constructor(private deps: BlockAndLogStreamerListenerDependencies) {
     this.address = deps.address;
     deps.blockAndLogStreamer.subscribeToOnLogsAdded(this.onLogsAdded);
   }
 
-  public static create(provider: EthersProvider, eventLogDBRouter: EventLogDBRouter, address: string, getEventTopics: ((eventName: string) => Array<string>)) {
+  static create(
+    provider: EthersProvider,
+    eventLogDBRouter: EventLogDBRouter,
+    address: string,
+    getEventTopics: ((eventName: string) => string[])
+  ) {
     const dependencies = new EthersProviderBlockStreamAdapter(provider);
-    const blockAndLogStreamer = new BlockAndLogStreamer<Block, ExtendedLog>(dependencies.getBlockByHash, dependencies.getLogs, (error: Error) => {
-      console.error(error);
-    });
+    const blockAndLogStreamer = new BlockAndLogStreamer<Block, ExtendedLog>(
+      dependencies.getBlockByHash,
+      dependencies.getLogs,
+      (error: Error) => {
+        console.error(error);
+      }
```

Comment:
You can just have the parameter here be `console.error`, no need to wrap it in a lambda.